proposals/0165-async-vote-execution.md

```diff
@@ -0,0 +1,218 @@
+---
+simd: '0165'
+title: Async Vote Execution
+authors:
+  - Wen Xu
+category: Standard
+type: Core
+status: Idea
+created: 2024-08-11
+feature: null
+supersedes: null
+superseded-by: null
+extends: null
+---
+
+## Summary
+
+Optimistically execute all vote transactions in a block to determine fork
+selection in consensus early on, before all the transactions in the block
+are fully executed and the actual fee payers for vote transactions are
+checked.
+
+This allows us to more quickly converge on one chain of blocks, so that
+validators don't have to execute any blocks not on selected fork. This saves
+CPU and memory resource needed in replay, it also ensures that the cluster
+will have fewer forks that are caused by slow transaction execution.
+
+## Motivation
+
+Currently the vote transactions and non-vote transactions are mixed together in
+a block, the vote transactions are only processed in consensus when the whole
+block has been frozen and all transactions in the block have been verified and
```

Comment:
nit: votes are processed per entry batch, doesn't have to wait for the bank to be frozen. as a result we still partially process votes even if the block later ends up dead.
e.g. agave's impl https://github.com/anza-xyz/agave/blob/1f06fbdbe3b72f330f50ad93a15c1116f5021392/ledger/src/blockstore_processor.rs#L177

Reply:
Yeah, now I think about it, it's less about processing votes as fast as possible, more about a lot of things don't happen until the block is frozen, changes this part.